echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/eventhandlers/ManualEventMonitor.java

```diff
@@ -40,48 +37,42 @@
  */
 @Slf4j
 @Component
-public class ManualEventMonitor extends TriggerMonitor {
-
+public class ManualEventMonitor implements TriggerEventHandler<ManualEvent> {
   public static final String MANUAL_TRIGGER_TYPE = "manual";
 
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
   @Override
-  protected boolean handleEventType(String eventType) {
+  public boolean handleEventType(String eventType) {
     return eventType.equalsIgnoreCase(ManualEvent.TYPE);
   }
 
   @Override
-  protected ManualEvent convertEvent(Event event) {
+  public ManualEvent convertEvent(Event event) {
     return objectMapper.convertValue(event, ManualEvent.class);
   }
 
-  public ManualEventMonitor(@NonNull PipelineCache pipelineCache,
-    @NonNull PipelineInitiator pipelineInitiator,
-    @NonNull Registry registry) {
-    super(pipelineCache, pipelineInitiator, registry);
+  public List<Pipeline> getMatchingPipelines(ManualEvent manualEvent, List<Pipeline> pipelines) {
+    return pipelines.stream()
+      .map(p -> withMatchingTrigger(manualEvent, p))
```

Comment:
Just curious, why isn't this logic in a base class? It feels like this "find & return matching pipelines" will be shared by all event monitors.

Reply:
+1

Reply:
I think the similarity here is a vestige of when `ManualEventMonitor` had to be a subclass of `TriggerMonitor` (so that it did not need to re-implement all the triggering logic) and thus needed to unnaturally match the paradigm of the other monitors.  It's somewhat different than the other monitors, as it doesn't look at pipeline triggers at all---it just finds whatever pipeline matches the id it's supposed to execute.  I've refactored it to be much clearer now that it doesn't need to follow the pattern of the others.

That being said, it's true that every handler is going to loop over pipelines---so I've pushed the looping part to the base class and have made the implementations just handle deciding if a particular pipeline matches.